container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go

```diff
@@ -1482,6 +1484,105 @@ func (manager *KubernetesManager) RunExecCommand(
+	podName string,
+	containerName string,
+	command []string,
+) (chan string, chan *exec_result.ExecResult) {
+	execOutputChan := make(chan string)
+	finalExecResultChan := make(chan *exec_result.ExecResult)
+	go func() {
+		defer func() {
+			close(execOutputChan)
+			close(finalExecResultChan)
+		}()
+		execOptions := &apiv1.PodExecOptions{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "",
+				APIVersion: "",
+			},
+			Stdin:     false,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       true,
+			Container: containerName,
+			Command:   command,
+		}
+
+		//Create a RESTful command request.
+		request := manager.kubernetesClientSet.CoreV1().RESTClient().
+			Post().
+			Namespace(namespaceName).
+			Resource("pods").
+			Name(podName).
+			SubResource("exec").
+			VersionedParams(execOptions, scheme.ParameterCodec)
+		if request == nil {
+			sendErrorAndFail(execOutputChan, stacktrace.NewError(
+				"Failed to build a working RESTful request for the command '%s'.",
+				execOptions.Command),
+				"An error occurred while streaming from kubernetes with following exit code '%d'",
+				-1)
+			return
+		}
+
+		exec, err := remotecommand.NewSPDYExecutor(manager.kuberneteRestConfig, http.MethodPost, request.URL())
+		if err != nil {
+			sendErrorAndFail(
+				execOutputChan,
+				stacktrace.Propagate(err,
+					"Failed to build an executor for the command '%s' with the RESTful endpoint '%s'.", execOptions.Command, request.URL().String()), ""+
+					"An error occurred while streaming from kubernetes with following exit code '%d'",
+				-1)
+			return
+		}
+
+		// Stream output from k8s to output channel
+		outputBuffer := &bytes.Buffer{}
+		streamWriter := stream_writer.NewStreamWriter(outputBuffer, execOutputChan)
+		if err = exec.StreamWithContext(ctx, remotecommand.StreamOptions{
+			Stdin:             nil,
+			Stdout:            streamWriter,
+			Stderr:            streamWriter,
+			Tty:               true,
+			TerminalSizeQueue: nil,
+		}); err != nil {
+			// Kubernetes returns the exit code of the command via a string in the error message, so we have to extract it
+			statusError := err.Error()
+
+			// this means that context deadline has exceeded
+			if strings.Contains(statusError, contextDeadlineExceeded) {
+				sendErrorAndFail(
+					execOutputChan,
+					stacktrace.Propagate(err, "There was an error occurred while executing commands on the container"),
+					"An error occurred while streaming from kubernetes with following exit code '%d'",
+					1)
+				return
+			}
+
+			exitCode, err := getExitCodeFromStatusMessage(statusError)
+			if err != nil {
+				sendErrorAndFail(
+					execOutputChan,
+					stacktrace.Propagate(err, "There was an error trying to parse the message '%s' to an exit code.", statusError),
+					"An error occurred while streaming from kubernetes with following exit code '%d'",
+					1)
+				return
+			}
+
+			finalExecResultChan <- exec_result.NewExecResult(exitCode, outputBuffer.String())
```

Comment:
Still wondering why you need to send the entire output once again here. If you can avoid it, then you basically don't need your StreamWriter